hub/management/commands/import_mps_appg_data.py

```diff
@@ -0,0 +1,127 @@
+    data_url = get_dataset_url(
+            repo_name="mp-appg-membership-data",
+            package_name="mp_appg_membership_data",
+            version_name="latest",
+            file_name="appg_officers.csv",
+        )
+    
+    def handle(self, quiet=False, *args, **options):
+        self._quiet = quiet
+        self.import_results()
+
+    def add_mps(self, df):
+        # Adds the MPs as Person objects to the df - note that this sill not add all
+        # of the officers in the df, because some of them are in the house of Lords,
+        # rather than are MPs.
+        twfy_ids = PersonData.objects.filter(data_type__data_set__name="twfyid")
+        df.twfy_id = df.twfy_id.str[25:].astype(int)
+        df["mp"] = df.twfy_id.dropna().apply(lambda x: twfy_ids.filter(data=x).first().person if twfy_ids.filter(data=x) else None)
+        df = df.dropna(subset='mp')
+        return df
+
+    def get_df(self):
+        df = get_dataset_df(
+            repo_name="mp-appg-membership-data",
+            package_name="mp_appg_membership_data",
+            version_name="latest",
+            file_name="appg_officers.csv",
+        )
+        # Limit the df to only the APPGs relating to climate
+        # (Currently, this is done by reading from a list produced
+        # by searching for various keywords related to climate in the
+        # title and purposes of the APPGs)
+        with open("data/climate_APPGs.txt", "r") as fh:
+            climate_appgs = [line.replace("\n", "") for line in fh.readlines()]
+        df = df[df.appg_name.isin(climate_appgs)]
+        df = self.add_mps(df)
+        return df
+
+    def get_df(self):
+        df = get_dataset_df(
+            repo_name="mp-appg-membership-data",
+            package_name="mp_appg_membership_data",
+            version_name="latest",
+            file_name="appg_officers.csv",
+        )
+        # Limit the df to only the APPGs relating to climate
+        # (Currently, this is done by reading from a list produced
+        # by searching for various keywords related to climate in the
+        # title and purposes of the APPGs)
+        with open("data/climate_APPGs.txt", "r") as fh:
+            climate_appgs = [line.replace("\n", "") for line in fh.readlines()]
+        df = df[df.appg_name.isin(climate_appgs)]
+        df = self.add_mps(df)
+        return df
+
+    def create_data_type(self):
+        appg_membership_ds, created = DataSet.objects.update_or_create(
+            name="mp_appg_memberships",
+            defaults={
+                "data_type": "text",
+                "description": "Membership in APPGs as published on the parliament website",
+                "source": "https://parliament.uk/",
+                "table": "person__persondata",
+            },
+        )
+
+        appg_membership, created = DataType.objects.update_or_create(
+            data_set=appg_membership_ds,
+            name="appg_membership",
```

Comment:
This should be the same name ("mp_appg_memberships") as the dataset otherwise the filtering doesn't work.